src/sentry/issues/suspect_flags.py

```diff
@@ -0,0 +1,126 @@
+from collections import defaultdict
+from collections.abc import Mapping
+from datetime import datetime
+
+from snuba_sdk import Column, Condition, Entity, Function, Op, Query, Request
+
+from sentry.seer.workflows.compare import kl_compare_sets
+from sentry.utils.snuba import raw_snql_query
+
+Attributes = Mapping[str, dict[str, float]]
+AttributesRow = tuple[str, str, float]
+Score = tuple[str, float]
+
+
+def get_suspect_flag_scores(
+    org_id: int, project_id: int, start: datetime, end: datetime, primary_hash: str
+) -> list[Score]:
+    """
+    Queries the baseline and outliers sets. Computes the KL scores of each and returns a sorted
+    list of key, score values.
+    """
+    baseline_rows = query_flag_rows(org_id, project_id, start, end, primary_hash=None)
+    baseline = as_attribute_dict(baseline_rows)
+
+    outliers_rows = query_flag_rows(org_id, project_id, start, end, primary_hash=primary_hash)
+    outliers = as_attribute_dict(outliers_rows)
+
+    return kl_score(baseline, outliers)
+
+
+def query_flag_rows(
+    organization_id: int,
+    project_id: int,
+    start: datetime,
+    end: datetime,
+    primary_hash: str | None,
+) -> list[AttributesRow]:
+    """
+    Query for the count of unique flag-key, flag-value pairings. Specifying a primary-hash will
+    narrow the query to a particular issue.
+
+    SQL:
+        SELECT arrayJoin(arrayZip(flags.key, flags.value)) as variants, count()
+        FROM errors_dist
+        WHERE (
+            project_id = {project_id} AND
+            timestamp >= {start} AND
+            timestamp < {end} AND
+            primary_hash = {primary_hash}
```

Comment:
Should we be using issue id instead? Using primary hash means that this won't handle merges correctly

Reply:
I didn't realize `group_id` was the issue id.  Thanks I'll use that instead.